nan_json.h

```diff
@@ -0,0 +1,160 @@
+#endif  // NODE_MODULE_VERSION > NODE_6_0_MODULE_VERSION
+
+class JSON {
+ public:
+  JSON() {
+#if NAN_JSON_H_NEED_PARSE + NAN_JSON_H_NEED_STRINGIFY
+    Nan::MaybeLocal<v8::Value> maybe_global_json = Nan::Get(
+      Nan::GetCurrentContext()->Global(),
+      Nan::New("JSON").ToLocalChecked()
+    );
+
+    if (!maybe_global_json.IsEmpty()) {
+      v8::Local<v8::Value> val_global_json = maybe_global_json.ToLocalChecked();
+
+      if (val_global_json->IsObject()) {
+        Nan::MaybeLocal<v8::Object> maybe_obj_global_json =
+          Nan::To<v8::Object>(val_global_json);
+
+        if (!maybe_obj_global_json.IsEmpty()) {
+          v8::Local<v8::Object> global_json =
+            maybe_obj_global_json.ToLocalChecked();
+
+#if NAN_JSON_H_NEED_PARSE
+          Nan::MaybeLocal<v8::Value> maybe_parse_method = Nan::Get(
+            global_json, Nan::New("parse").ToLocalChecked()
+          );
+
+          if (!maybe_parse_method.IsEmpty()) {
+            v8::Local<v8::Value> parse_method =
+              maybe_parse_method.ToLocalChecked();
+
+            if (parse_method->IsFunction()) {
+              parse_cb_.Reset(parse_method.As<v8::Function>());
+            }
+          }
+#endif  // NAN_JSON_H_NEED_PARSE
+
+#if NAN_JSON_H_NEED_STRINGIFY
+          Nan::MaybeLocal<v8::Value> maybe_stringify_method = Nan::Get(
+            global_json, Nan::New("stringify").ToLocalChecked()
+          );
+
+          if (!maybe_stringify_method.IsEmpty()) {
+            v8::Local<v8::Value> stringify_method =
+              maybe_stringify_method.ToLocalChecked();
+
+            if (stringify_method->IsFunction()) {
+              stringify_cb_.Reset(stringify_method.As<v8::Function>());
+            }
+          }
+#endif  // NAN_JSON_H_NEED_STRINGIFY
+        }
+      }
+    }
+#endif  // NAN_JSON_H_NEED_PARSE + NAN_JSON_H_NEED_STRINGIFY
+  }
+
+  inline
+  Nan::MaybeLocal<v8::Value> Parse(v8::Local<v8::String> json_string) {
+    Nan::EscapableHandleScope scope;
+#if NAN_JSON_H_NEED_PARSE
+    return scope.Escape(parse(json_string));
+#else
+#if NODE_MODULE_VERSION > NODE_6_0_MODULE_VERSION
+    Nan::MaybeLocal<v8::Value> result =
+      v8::JSON::Parse(Nan::GetCurrentContext(), json_string);
+
+    if (result.IsEmpty()) return v8::Local<v8::Value>();
+    return scope.Escape(result.ToLocalChecked());
+#else
+    return scope.Escape(v8::JSON::Parse(json_string));
+#endif  // NODE_MODULE_VERSION > NODE_6_0_MODULE_VERSION
+#endif  // NAN_JSON_H_NEED_PARSE
+  }
+
+  inline
+  Nan::MaybeLocal<v8::String> Stringify(v8::Local<v8::Object> json_object) {
+    Nan::EscapableHandleScope scope;
+    Nan::MaybeLocal<v8::String> result =
+#if NAN_JSON_H_NEED_STRINGIFY
+      Nan::To<v8::String>(stringify(json_object));
+#else
+      v8::JSON::Stringify(Nan::GetCurrentContext(), json_object);
+#endif  // NAN_JSON_H_NEED_STRINGIFY
+    if (result.IsEmpty()) return v8::Local<v8::String>();
+    return scope.Escape(result.ToLocalChecked());
```

Comment:
This was inspired by your earlier comment, https://github.com/nodejs/nan/pull/651#issuecomment-287055725 ...  I'll be happy to change it in any way you prefer.  I don't have any preference on this one, so I'll leave it unless told otherwise.